src/main/java/com/amazonaws/samples/S3Sample.java

```diff
@@ -0,0 +1,215 @@
+        Region usWest2 = Region.getRegion(Regions.US_WEST_2);
+        s3.setRegion(usWest2);
+
+        String bucketName = "my-first-s3-bucket-" + UUID.randomUUID();
+        String key = "MyObjectKey";
+
+        System.out.println("===========================================");
+        System.out.println("Getting Started with Amazon S3");
+        System.out.println("===========================================\n");
+
+        try {
+            /*
+             * Create a new S3 bucket - Amazon S3 bucket names are globally unique,
+             * so once a bucket name has been taken by any user, you can't create
+             * another bucket with that same name.
+             *
+             * You can optionally specify a location for your bucket if you want to
+             * keep your data closer to your applications or users.
+             */
+            System.out.println("Creating bucket " + bucketName + "\n");
+            s3.createBucket(bucketName);
+
+            /*
+             * List the buckets in your account
+             */
+            System.out.println("Listing buckets");
+            for (Bucket bucket : s3.listBuckets()) {
+                System.out.println(" - " + bucket.getName());
+            }
+            System.out.println();
+
+            /*
+             * Upload an object to your bucket - You can easily upload a file to
+             * S3, or upload directly an InputStream if you know the length of
+             * the data in the stream. You can also specify your own metadata
+             * when uploading to S3, which allows you set a variety of options
+             * like content-type and content-encoding, plus additional metadata
+             * specific to your applications.
+             */
+            System.out.println("Uploading a new object to S3 from a file\n");
+            s3.putObject(new PutObjectRequest(bucketName, key, createSampleFile()));
+
+            /*
+             * Download an object - When you download an object, you get all of
+             * the object's metadata and a stream from which to read the contents.
+             * It's important to read the contents of the stream as quickly as
+             * possibly since the data is streamed directly from Amazon S3 and your
+             * network connection will remain open until you read all the data or
+             * close the input stream.
+             *
+             * GetObjectRequest also supports several other options, including
+             * conditional downloading of objects based on modification times,
+             * ETags, and selectively downloading a range of an object.
+             */
+            System.out.println("Downloading an object");
+            S3Object object = s3.getObject(new GetObjectRequest(bucketName, key));
+            System.out.println("Content-Type: "  + object.getObjectMetadata().getContentType());
+            displayTextInputStream(object.getObjectContent());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getObjectContent*. Currently, there are execution paths that do not contain closure statements. Either a) close the object returned by *getObjectContent()* in a try-finally block or b) close the resource by declaring the object returned by *getObjectContent()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).